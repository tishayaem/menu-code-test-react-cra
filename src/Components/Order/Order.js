import React from 'react';
import { formatPrice } from '../../helpers';
export default function Order(props) {
  let total;
  if (Object.keys(props.currentOrder).length !== 0) {
    total = Object.values(props.currentOrder).reduce((total, value) => {
      return total + value.price;
    }, 0);
  } else {
    total = 0;
  }

  let totalForTwo = Object.values(props.order).reduce(
    (total, value) =>
      total +
      Object.values(value).reduce((total, value) => {
        return total + value.price;
      }, 0),
    0
  );

  return (
    <React.Fragment>
      <div>
      {props.user < 3 ? <div className="current-order"><h3>Order for Guest {props.user}</h3><label>{formatPrice(total)}</label> </div> : null}
      
      
      </div>
      <div className="final-order">
      <h3>Total amount</h3>
      <label>{formatPrice(totalForTwo)}</label>
      </div>
    </React.Fragment>
  );
}

