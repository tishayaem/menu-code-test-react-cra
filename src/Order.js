import React from 'react';
import { formatPrice } from './helpers';
export default function Order(props) {
  let total;
  if (Object.keys(props.currentOrder).length !== 0) {
    total = Object.values(props.currentOrder).reduce((total, value) => {
      return total + value.price;
    }, 0);
  } else {
    total = 0;
  }

  return <div className="current-order">{formatPrice(total)} </div>;
}


