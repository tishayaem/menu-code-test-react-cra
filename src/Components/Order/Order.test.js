import React from 'react';
import Order from './Order';

describe('Order', () => {
  // Move dummy data to a separate file
  const currentOrder = {
    starters: {
      id: 1,
      name: 'Borscht',
      price: 5
    },
    mains: {
      id: 2,
      name: 'Vareniki',
      price: 10
    },
    desserts: {
      id: 3,
      name: 'Mille-feuille',
      price: 5
    }
  };
  const order = {
    user1: {
      starters: {
        id: 1,
        name: 'Borscht',
        price: 5
      },
      mains: {
        id: 2,
        name: 'Vareniki',
        price: 10
      },
      desserts: {
        id: 3,
        name: 'Mille-feuille',
        price: 5
      }
    }
  };
  const wrapper = shallow(<Order currentOrder={currentOrder} order={order} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct message', () => {
    expect(wrapper.find('.final-order').text()).toMatch('Â£20.00');
  });
});
