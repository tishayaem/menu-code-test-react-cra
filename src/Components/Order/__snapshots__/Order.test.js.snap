// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Order
    currentOrder={
      Object {
        "desserts": Object {
          "id": 3,
          "name": "Mille-feuille",
          "price": 5,
        },
        "mains": Object {
          "id": 2,
          "name": "Vareniki",
          "price": 10,
        },
        "starters": Object {
          "id": 1,
          "name": "Borscht",
          "price": 5,
        },
      }
    }
    order={
      Object {
        "user1": Object {
          "desserts": Object {
            "id": 3,
            "name": "Mille-feuille",
            "price": 5,
          },
          "mains": Object {
            "id": 2,
            "name": "Vareniki",
            "price": 10,
          },
          "starters": Object {
            "id": 1,
            "name": "Borscht",
            "price": 5,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div />,
        <div
          className="final-order"
        >
          <h3>
            Total amount
          </h3>
          <label>
            £20.00
          </label>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Total amount
            </h3>,
            <label>
              £20.00
            </label>,
          ],
          "className": "final-order",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total amount",
            },
            "ref": null,
            "rendered": "Total amount",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "£20.00",
            },
            "ref": null,
            "rendered": "£20.00",
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div />,
          <div
            className="final-order"
          >
            <h3>
              Total amount
            </h3>
            <label>
              £20.00
            </label>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Total amount
              </h3>,
              <label>
                £20.00
              </label>,
            ],
            "className": "final-order",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total amount",
              },
              "ref": null,
              "rendered": "Total amount",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "£20.00",
              },
              "ref": null,
              "rendered": "£20.00",
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
